lib/collection/src/shards/replica_set/update.rs

```diff
@@ -241,12 +293,18 @@ impl ShardReplicaSet {
                 let echo_tag = result.clock_tag?;
 
                 if echo_tag.peer_id != clock_tag.peer_id {
-                    // TODO: `log::warn`!?
+                    debug_assert!(
+                        false,
+                        "Echoed clock tag peer_id does not match the original"
+                    );
```

Comment:
```suggestion
                    #[cfg(debug_assertions)]
                    panic!("Echoed clock tag peer_id does not match the original");
```

Not sure if this is better, though.

Reply:
Nevermind. With `panic` it starts complaining about `return None` being unreachable. :/